tutorials/dataframe/df106_HiggsToFourLeptons_withVariations.py

```diff
@@ -81,25 +92,26 @@
 df = df.Define("goodlep_pt", "lep_pt[good_lep]")\
        .Define("goodlep_eta", "lep_eta[good_lep]")\
        .Define("goodlep_phi", "lep_phi[good_lep]")\
-       .Define("goodlep_E", "lep_E[good_lep]")
+       .Define("goodlep_E", "lep_E[good_lep]")\
+       .Define("goodlep_type", "lep_type[good_lep]")
 
 # Select leptons with high transverse momentum
 df = df.Filter("goodlep_pt[0] > 25000 && goodlep_pt[1] > 15000 && goodlep_pt[2] > 10000")
 
 # Reweighting of the samples is different for "data" and "MC". This is the function to add reweighting for MC samples
 ROOT.gInterpreter.Declare("""
-double weights(float scaleFactor_1, float scaleFactor_2, float scaleFactor_3, float scaleFactor_4, float mcWeight_, double xsecs_, double sumws_, double lumi_)
+double weights(float scaleFactor_1, float scaleFactor_2, float scaleFactor_3, float scaleFactor_4, float scale, float mcWeight_, double xsecs_, double sumws_, double lumi_)
 {
-    double weight_ = scaleFactor_1 * scaleFactor_2 * scaleFactor_3 * scaleFactor_4 * mcWeight_ * xsecs_/sumws_ * lumi_;
+    double weight_ = scaleFactor_1 * scaleFactor_2 * scaleFactor_3 * scaleFactor_4 * scale * mcWeight_ * xsecs_/sumws_ * lumi_;
     return (weight_);
 }
 """)
 
 # Use DefinePerSample to define which samples need reweighting
 df = df.DefinePerSample("reweighting", 'rdfsampleinfo_.Contains("mc")')
-df = df.Define("weight", 'double x; if (reweighting) {return weights(scaleFactor_ELE, scaleFactor_MUON, scaleFactor_LepTRIGGER, scaleFactor_PILEUP, mcWeight, xsecs, sumws, lumi);} else {return 1.;}')
+df = df.Define("weight", 'double x; if (reweighting) {return weights(scaleFactor_ELE, scaleFactor_MUON, scaleFactor_LepTRIGGER, scaleFactor_PILEUP, scale, mcWeight, xsecs, sumws, lumi);} else {return 1.;}')
```

Comment:
You can again use the ternary operator with the `... ? ... : ...` here if you want